src/shims/native_lib/mod.rs

```diff
@@ -226,46 +217,56 @@ trait EvalContextExtPriv<'tcx>: crate::MiriInterpCxExt<'tcx> {
         let this = self.eval_context_mut();
 
         for evt in events.acc_events {
-            //eprintln!("evt: {evt:?}");
             let evt_rg = evt.get_range();
             // LLVM at least permits vectorising accesses to adjacent allocations,
             // so we cannot assume 1 access = 1 allocation. :(
             let mut rg = evt_rg.addr..evt_rg.end();
             while let Some(curr) = rg.next() {
-                let alloc_id = this
-                    .alloc_id_from_addr(curr.to_u64(), rg.len().try_into().unwrap(), true)
-                    .expect("Foreign code did an out-of-bounds access!");
+                let Some(alloc_id) =
+                    this.alloc_id_from_addr(curr.to_u64(), rg.len().try_into().unwrap(), true)
+                else {
+                    throw_ub_format!("Foreign code did an out-of-bounds access!")
+                };
                 let alloc = this.get_alloc_raw(alloc_id)?;
                 let alloc_addr = alloc.get_bytes_unchecked_raw().addr();
 
                 // Skip forward however many bytes of the access are contained in the current allocation.
+                // The start of the overlap range will be the greater of the alloc base address and the
+                // lowest remaining address in the access' range; the end will be the lesser of the end
```

Comment:
Isn't the start of the overlap always `curr` since we go left-to-right?

Reply:
Fair point!